cpp/include/cuspatial/detail/utility/offset_to_keys.cuh

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <thrust/binary_search.h>
+#include <thrust/distance.h>
+
+/** @brief Given list offset and row `i`, return a unique key that represent the list of `i`.
```

Comment:
Why is this a "unique key"? It's just the index at which to insert `i` into the offset array to keep it sorted.  It's not unique, as your example shows: both 1 and 2 map to index 4.